### PR TITLE
Closes #1979 Add a flag_all function argument to enhance `derive_var_extreme_flag()`

### LINKED ISSUE #1979: Feature Request: Add option to flag all extreme records to `derive_var_extreme_flag()`
### Feature Idea

The `flag_all` argument should be added to `derive_var_extreme_flag()`. The default is `FALSE`. If it is set to `TRUE`, all records are flagged and no error or warning is issued if the first or last record is not unique.

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

```
derive_var_extreme_flag(
      by_vars = exprs(STUDYID, USUBJID, BASETYPE, AVISITN),
      order = exprs(ADTM),
      new_var = LAST02FL,
      mode = "last",
      flag_all = TRUE
)
```
This would flag at each visit the last assessment. If this is not unique, all records are flagged, e.g., there could be several parameters at the last date.

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: tests/testthat/test-derive_var_extreme_flag.R [hunk 1]
```diff
@@ -27,7 +27,8 @@ input_worst_flag <- tibble::tribble(
   "TEST01", "PAT02", "PARAM03", "BASELINE", as.Date("2021-04-30"), 12.0
 )
 
-test_that("first observation for each group is flagged", {
+## Test 1: first observation for each group is flagged ----
```
Comment (bms63) [inline tests/testthat/test-derive_var_extreme_flag.R:30]: Thanks for updated these to follow our test guidance.  Is the big tibble above this used at all in these tests?  If not, can we delete it?

**Comment by bms63** [comment]
Could we get a test or two with some dirtier data?  Like a subject missing an AVISITN or missing AVAL?

**Comment by bms63** [comment]
Should we mention this `flag_all` in the Details section as well?

File: tests/testthat/test-derive_var_extreme_flag.R (hunk 1 above)
Comment (zdz2101) [reply tests/testthat/test-derive_var_extreme_flag.R:30]: I assumed it was in the previous tests but I suppose its actually not, going to go ahead and remove it

**Comment by zdz2101** [comment]
> Should we mention this `flag_all` in the Details section as well?

I can add some flavoring

**Review by bms63 — APPROVED** [review]
This looks good to me now. @bundfussr or @manciniedoardo any thoughts?

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
LGTM, only comment I have is that we could maybe showcase the `flag_all = TRUE` case in one of our header examples? Eg, in the example starting on line 121, change to "Most severe AE first occurrence per patient (flag all cases)" and then remove `ASEQ` from `by_vars` and add `flag_all = TRUE` as an argument. Then the first two "MILD" AEs for the first patient would be flagged, showcasing the functionality nicely.

File: R/derive_var_extreme_flag.R [hunk 2]
```diff
@@ -26,6 +26,11 @@
 #'
 #'   Permitted Values: `"first"`, `"last"`
 #'
+#' @param flag_all Flag setting
+#'
+#'   A logical value where if set to TRUE, all records are flagged
```
Comment (bundfussr) [inline R/derive_var_extreme_flag.R:31]: Please add backticks for `TRUE`.

File: R/derive_var_extreme_flag.R [hunk 3]
```diff
@@ -242,6 +260,21 @@ derive_var_extreme_flag <- function(dataset,
...
+    if (mode == "first") {
+      data <- data %>%
+        group_by(!!!by_vars, !!!order) %>%
+        fill(!!new_var, .direction = "down") %>%
+        ungroup()
+    } else {
+      data <- data %>%
+        group_by(!!!by_vars, !!!order) %>%
+        fill(!!new_var, .direction = "up") %>%
+        ungroup()
```
Comment (bundfussr) [inline R/derive_var_extreme_flag.R:273]: Please avoid duplication of code.

File: tests/testthat/test-derive_var_extreme_flag.R [hunk 4]
```diff
@@ -80,3 +54,138 @@ test_that("last observation for each group is flagged", {
...
+    test_flag_all,
+    by_vars = exprs(STUDYID, USUBJID, BASETYPE, AVISITN),
+    order = exprs(ADTM),
+    new_var = FIRSTFL,
+    mode = "first",
+    check_type = "none",
+    flag_all = FALSE
+  )
+
+  expect_true(all.equal(expected_output, actual_output))
```
Comment (bundfussr) [inline tests/testthat/test-derive_var_extreme_flag.R:83]: Please use `expect_dfs_equal()`.

**Review by bundfussr — CHANGES_REQUESTED** [review]

**Review by bundfussr — APPROVED** [review]
